synapse/storage/background_updates.py

```diff
@@ -533,6 +533,7 @@ def register_background_index_update(
             index_name: name of index to add
             table: table to add index to
             columns: columns/expressions to include in index
+            where_clause: A where clause to apply to the index.
```

Comment:
Should this also use the phrase "index predicate"?

Reply:
I suspect it should say "partial index" somewhere, does this sound OK?

```suggestion
            where_clause: A where clause to create a partial index.
```

Reply:
Not sure about "create"---I think a unique index across the given columns and with the given index predicate has to already exist for the query to make sense?

How about
```suggestion
where_clause: A WHERE clause to specify a partial unique index.
```

The phrase "partial unique index" seems to be pretty searchable: https://www.postgresql.org/docs/current/indexes-partial.html is the top hit for me!